text/0039-_formatted-field-behavior.md

```diff
@@ -0,0 +1,473 @@
+    ]
+}
+```
+
+`_formatted` is only filled with the `attributesToRetrieve` fields despite the fact that the user may wants all fields to be in `_formatted` and be highlighted or cropped given `attributesToHighlight` or `attributesToCrop` values.
+
+#### Current Transplant/Milli Behavior (0.21)
+
+**Example 1**
+
+Given these search parameters:
+```
+{
+    "q": "prince",
+    "attributesToRetrieve": ["*"]
+}
+```
+
+As a user i get:
+
+```
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "actor": "Prince",
+            "poster": "https://image.tmdb.org/t/p/w1280/3KHiQt54usbHyIjLIMzaDAoIJNK.jpg"
+        }
+    ]
+}
+```
+
+Since `attributesToHighlight` and `attributesToCrop` are not set, `_formatted` is not computed.
+
+Same behavior as v0.20 release.
+
+**Example 2**
+
+Given these search parameters:
+```
+{
+    "q": "prince",
+    "attributesToRetrieve": ["*"],
+    "attributesToHighlight": ["wrongFieldName"]
+}
+```
+
+As a user i get:
+
+```
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "poster": "https://image.tmdb.org/t/p/w1280/3KHiQt54usbHyIjLIMzaDAoIJNK.jpg",
+        }
+    ]
+}
+```
+
+Unlike v0.20, if no field are matched to be formatted, `_formatted` is not computed nor returned in response.
+
+**Example 3**
+
+Given these search parameters:
+
+```
+{
+    "q": "Prince",
+    "attributesToRetrieve": ["title"],
+    "attributesToHighlight": ["actor"]
+}
+```
+
+As a user i get:
+
+```
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "_formatted": {
+                "actor": "<em>Prince</em>"
+            }
+        }
+    ]
+}
+```
+
+`_formatted` is only filled with the fields set in `attributesToHighlight` despite the fact that the user only ask for `title` in `attributesToRetrieve`.
+
+Same behavior as v0.20 release.
+
+**Example 4**
+
+Given these search paramters:
+
+```
+{
+    "q": "Prince",
+    "attributesToRetrieve": ["actor", "title"],
+    "attributesToHighlight": ["actor"]
+}
+```
+
+As a user i get:
+
+```
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "actor": "Prince",
+            "_formatted": {
+                "actor": "<em>Prince</em>"
+            }
+        }
+    ]
+}
+```
+
+Same behavior as v0.20 release.
+
+**Example 5**
+
+Given these search parameters:
+```
+{
+    "q": "prince",
+    "attributesToRetrieve": ["*"],
+    "attributesToHighlight": ["title"]
+}
+```
+
+As a user i get:
+
+```
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "actor": "Prince",
+            "poster": "https://image.tmdb.org/t/p/w1280/3KHiQt54usbHyIjLIMzaDAoIJNK.jpg",
+            "_formatted": {
+                "title": "<em>Prince</em> Avalanche"
+            }
+        }
+    ]
+}
+```
+
+Unlike v0.20, `_formatted` is only containing fields set in `attributesToHighlight` and `attributesToCrop`. Independently from `attributesToRetrieve` value.
+
```

Comment:
This means, contrary to the v0.20.0, there is no way for the front-end developer to have all the attributes (highlighted and non-highlighted) without merging `_formatted` with the other attributes. It's not complicated to do, but we clearly lose dev experience for front-end developers with this choice. I remember charlotte talked about this with you.
Is it what we want?

Reply:
This is the current behavior of 0.21. I've made a proposal to change that and have the behavior that you explain here. This is in the "Expected MeiliSearch Behavior" part below. Line 402 for an explanation and Line 432 to 459 for an example.

Reply:
I miss read! Sorry!! I thought it was "what you expect for the v0.21.0" 😁